runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -566,10 +576,11 @@ def ssh(self, node: str = None):
                 ssh_private_key=str(sky_key),
                 docker_user=self.docker_user,
             )
-            cmd = runner.run(
-                cmd="bash --rcfile <(echo '. ~/.bashrc; conda deactivate')",
-                ssh_mode=SshMode.INTERACTIVE,
-                port_forward=None,
-                return_cmd=True,
+            ssh_command = runner._ssh_base_command(
+                ssh_mode=SshMode.INTERACTIVE, port_forward=None
+            )
+            subprocess.run(
+                " ".join(ssh_command),
```

Comment:
Should this be a shlex join or is it already properly escaped from inside the command runner?